### PR TITLE
Introduce `FormatterSettings`

### PR DESCRIPTION
**Pull request by MichaReiser**
## Stack Summary

This stack splits `Settings` into `FormatterSettings` and `LinterSettings` and moves it into `ruff_workspace`. This change is necessary to add the `FormatterSettings` to `Settings` without adding `ruff_python_formatter` as a dependency to `ruff_linter` (and the linter should not contain the formatter settings). 

A quick overview of our settings struct at play:

* `Options`: 1:1 representation of the options in the `pyproject.toml` or `ruff.toml`.  Used for deserialization.
* `Configuration`: Resolved `Options`, potentially merged from multiple configurations (when using `extend`). The representation is very close if not identical to the `Options`.
* `Settings`: The resolved configuration that uses a data format optimized for reading. Optional fields are initialized with their default values. Initialized by `Configuration::into_settings` .

The goal of this stack is to split `Settings` into tool-specific resolved `Settings` that are independent of each other. This comes at the advantage that the individual crates don't need to know anything about the other tools. The downside is that information gets duplicated between `Settings`. Right now the duplication is minimal (`line-length`, `tab-width`) but we may need to come up with a solution if more expensive data needs sharing.

This stack focuses on `Settings`. Splitting `Configuration` into some smaller structs is something I'll follow up on later. 

## PR Summary

This is the final step for the settings related structure changes. This PR introduces a new `FormatterSettings` struct in `ruff_python_formatter` and populates it in `Configuration::into_settings`. This PR doesn't add new options yet (still only supports setting `line_width`).

## Test Plan

`cargo test`

* Run the hyperfine benchmark against `main` with caching. The numbers remain unchanged, proving that the cache is working
* Changing a setting that is not part of the cache key (e.g. output_format) does not result in a cache invalidation (same runtime as when running multiple times with the same settings)
* Changing any other setting results in invalidating the cache and recomputing the output.

### CODE REVIEW

**Comment by MichaReiser** [comment]
Current dependencies on/for this PR:
* main
  * **PR #7532** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/7532" target="_blank">Graphite</a> 
    * **PR #7542** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/7542" target="_blank">Graphite</a> 
      * **PR #7543** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/7543" target="_blank">Graphite</a> 
        * **PR #7544** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/7544" target="_blank">Graphite</a> 
          * **PR #7545** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/7545" target="_blank">Graphite</a>  👈
    * **PR #7536** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/7536" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/astral-sh/ruff/7545?utm_source=stack-comment).

File: crates/ruff_cli/src/commands/format.rs [hunk 1]
```diff
@@ -76,17 +73,7 @@ pub(crate) fn format(
...
-                    let preview = match resolved_settings.linter.preview {
-                        PreviewMode::Enabled => ruff_python_formatter::PreviewMode::Enabled,
-                        PreviewMode::Disabled => ruff_python_formatter::PreviewMode::Disabled,
-                    };
-                    let line_length = resolved_settings.linter.line_length;
-
-                    let options = PyFormatOptions::from_source_type(source_type)
-                        .with_line_width(LineWidth::from(NonZeroU16::from(line_length)))
-                        .with_preview(preview);
+                    let options = resolved_settings.formatter.to_format_options(source_type);
```
Comment (MichaReiser) [inline crates/ruff_cli/src/commands/format.rs:76]: All just for this 😆

Comment (charliermarsh) [reply crates/ruff_cli/src/commands/format.rs:76]: It's beautiful

**Review by charliermarsh — APPROVED** [review]
Excellent.

**Pull request merged**